### ISSUE TITLE
Add 5-Num Summary to top of boxplot

### ISSUE THREAD

**Issue by schanzer** [brownplt/code.pyret.org#331]
This is pretty urgent - coming up a lot in a DS PD we're leading right now. The box-plot (and pretty much ONLY the boxplot) is pretty damned useless once it's a static image. We desperately need to just spit out the 5-num summary on the image somehow.

**Comment by blerner** [comment]
After talking with Emanuel, this is _mostly_ moot for now, since he can finesse this in the BS:DS library itself.  But, I've worked on it somewhat today, and have two possible approaches, both of which are incomplete:

* I refactored the object in the `box-plot-series` to store a table of the 5-num summary info, that's currently being computed by [`get-box-data`](https://github.com/brownplt/code.pyret.org/blob/horizon/src/web/arr/trove/chart.arr#L683-L707).  This allows me to access that information as a `Table` (or as a 2d array, whatever, doesn't much matter) in the `_output` method of `box-plot-series`
* Approach 1: I can specialize the `_output` method of `box-plot-series` to return a `vs-seq` of the existing image and the `_output` of that summary `Table`.  This produces a manipulable rendered table of the summary info (can be copied, fractions can be toggled, etc).  We _might_ want to expose this as a function in the chart library, actually `get-summary(plot :: DataSeries%(is-box-plot-series)) -> Table`
* Approach 2: I can specialize the the `_output` method of `box-plot-series` to take the existing image and `beside` it with a rendered version of the same summary data.

Both of these approaches are incomplete in that they fail to display properly for `ChartWindow`s.  ChartWindows dont' have access to the underlying object; they just have a `render` closure that produces an image directly, which means I can't bypass it and get to the summary data.  I'm not sure what the best workaround here is.


(While working on this, I found several functions that are useful additions: `is-rectangular :: List<List<A>> -> Boolean` and `transpose :: List<List<A>>%(is-rectangular) -> List<List<A>>` are both good additions to the list libary, and `image-grid(imgs :: List<List<Image>>%(is-rectangular), padding-w :: Number, padding-h :: Number) -> Image` is a good addition to the image library.)
